superduperdb/jobs/job.py

```diff
@@ -151,7 +180,11 @@ def __call__(self, db: t.Union['Datalayer', None], dependencies=()):
         self.db = db
         db.metadata.create_job(self.dict())
 
-        self.submit(dependencies=dependencies)
+        if db.compute.remote is True:
+            self.submit_remote(self.identifier)
+        else:
+            self.submit(dependencies=dependencies)
```

Comment:
Why do we need this `db.compute.remote` - can just rewrite the `.submit` method of backends requiring special treatment?

Reply:
This is needed since submit function of RayBackend will run as a task and submit_remote will be a job submitter